stdlib/event/time.lua

```diff
@@ -0,0 +1,73 @@
+--- Time Event module
+-- @module Event.Time
+
+require 'stdlib/event/event'
+
+Event.Time = {}
+Event.Time._last_change = {}
+
+--All times are offset by 0.5
+--This is because both EvoGUI and MoWeather already apply that offset.
+--Following the precedent to remain consistent.
+--Actually, this little snippet is even borrowed from EvoGUI.
+if remote.interfaces.MoWeather then
+    -- assume MoWeather's getdaytime is sane
+    function Event.Time.get_day_time(surface_name_or_index) return remote.call("MoWeather", "getdaytime", surface_name_or_index) end
+else
+    -- 0.5 is midnight; let's make days *start* at midnight instead.
+    function Event.Time.get_day_time(surface_name_or_index) return game.surfaces[surface_name_or_index].daytime + 0.5 end
+end
+
+--These are Surface only events
+--- @field Gets around ldoc V1.4.3 eating the first field. :(
+Event.Time.ldoc_V1_4_3_eats_this   = nil
```

Comment:
Really? It was driving me nuts. I could rearrange the the `@field` events list however I wanted the first one always vanished. I could `grep -rin "ldoc_V1_4_3_eats_this" ./build/docs` and get no results. If I removed that from the lua file then the one that then became first would vanish from the generated docs. I was going batty I tell ya.

If it works for you I'll patch it out then. I actually forgot I had done that since I never saw it appear in the docs on my end. I made this part of the time event pretty early on.